CacheToolBundle.php

```diff
@@ -15,7 +15,47 @@
 use Symfony\Component\Finder\Finder;
 use Symfony\Component\HttpKernel\Bundle\Bundle;
 use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Bundle\FrameworkBundle\Console\Application as FrameworkApplication;
 
 class CacheToolBundle extends Bundle
 {
+    public function registerCommands(Application $application)
+    {
+        if (!$application instanceof FrameworkApplication) {
+            return parent::registerCommands($application);
+        }
+
+        if (!(false === ($apc = $application->getKernel()->getContainer()->getParameter('cachetool.apc'))
+            xor false === $application->getKernel()->getContainer()->getParameter('cachetool.opcache'))
+        ) {
+            return parent::registerCommands($application);
+        }
+
+        if (!is_dir($dir = $this->getPath().'/Command')) {
+            return;
+        }
+
+        $finder = new Finder();
+        $finder->files()->name(($apc ? 'Apc' : 'Opcache').'*Command.php')->in($dir);
```

Comment:
If you set `apc` and `opcache` to true in the configuration, the finder will only lookup for `Apc*Command`.